packages/nativescript-appdynamics/index.android.ts

```diff
@@ -40,6 +45,39 @@ export class Appdynamics implements IAppdynamics {
   }
 
   public requestTracker(url: string) {
-    return Instrumentation().beginHttpRequest(new java.net.URL(url));
+    return new RequestTracker(url);
+  }
+}
+
+export class RequestTracker implements IRequestTracker {
+  private _tracker;
+
+  constructor(url) {
+    this._tracker = Instrumentation().beginHttpRequest(new java.net.URL(url));
+  }
+
+  setError(error) {
+    this._tracker.withError(error);
+  }
+
+  reportDone(): void {
+    this._tracker.reportDone();
+  }
+
+  setHeaders(httpHeaders: HttpHeaders) {
+    const headerKeys = httpHeaders.keys();
+    const values: java.util.HashMap<string, java.util.List<string>> = new java.util.HashMap<string, java.util.List<string>>();
+
+    headerKeys.forEach((key) => {
+      const stringList: java.util.ArrayList<string> = new java.util.ArrayList<string>();
+      httpHeaders.getAll(key).forEach((headerValue) => stringList.add(headerValue));
```

Comment:
Is this secure? I believe this will also include any authentication headers and cookies, which would be a pretty serious security issue. It's probably better to work with a whitelist of headers to include.

Reply:
It doen't it just adds the header that you are passing into from your client application. So you can/should filter them in the client